src/v/cluster/metrics_reporter.cc

```diff
@@ -252,6 +255,9 @@ metrics_reporter::build_metrics_snapshot() {
     snapshot.has_oidc = config::oidc_is_enabled_kafka()
                         || config::oidc_is_enabled_http();
 
+    snapshot.has_rbac
+      = !_role_store.local().range([](auto const&) { return true; }).empty();
```

Comment:
I have a change on [this pr](https://github.com/redpanda-data/redpanda/pull/17152) that takes the `std::ranges` out of the `role_store` interface. 

So in this case we would wind up constructing a container of all role names to satisfy this query. For this usage, should we just add a `role_store::empty` method instead?

Other alternative is to leave the view in the interface, which might be preferable. That same PR breaks the near tree-wide dependency on `role_store.h` fwiw.

Reply:
> I have a change on [this pr](https://github.com/redpanda-data/redpanda/pull/17152) that takes the `std::ranges` out of the `role_store` interface. 
> 
> So in this case we would wind up constructing a container of all role names to satisfy this query. For this usage, should we just add a `role_store::empty` method instead?

Sufficient.